src/Model/Group.php

```diff
@@ -332,27 +380,29 @@ public static function expand($group_ids, $check_dead = false)
 	 * @param int    $gid   An optional pre-selected group
 	 * @param string $label An optional label of the list
 	 * @return string
-	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws \Exception
 	 */
 	public static function displayGroupSelection($uid, $gid = 0, $label = '')
 	{
-		$stmt = DBA::select('group', [], ['deleted' => 0, 'uid' => $uid], ['order' => ['name']]);
-
 		$display_groups = [
 			[
 				'name' => '',
 				'id' => '0',
 				'selected' => ''
 			]
 		];
+
+		$stmt = DBA::select('group', [], ['deleted' => 0, 'uid' => $uid], ['order' => ['name']]);
 		while ($group = DBA::fetch($stmt)) {
 			$display_groups[] = [
 				'name' => $group['name'],
 				'id' => $group['id'],
 				'selected' => $gid == $group['id'] ? 'true' : ''
 			];
 		}
-		Logger::log('groups: ' . print_r($display_groups, true));
+		DBA::close($stmt);
+
+		Logger::info('groups: ' . print_r($display_groups, true));
```

Comment:
Instead of adding ```print_r``` here, you could simply add the value as a parameter, means: ```Logger::info('Got groups', $display_groups);```

Reply:
Thanks, this allowed me to see I was using a method @nupplaphil removed (App->getDatabase).

Between the two of you, can we please agree on deprecating methods gracefully? That's twice in two days I had to scramble to revisit some code because of API drastic changes.

Reply:
*ack* .. I tried to get rid of the `App`->`get..()` methods because the class becomes more and more large .. I'll add it next time instead ..

Reply:
Standard deprecation procedure is to leave the method until the next release, then it can be removed if it isn't used anymore.

Reply:
Grrr https://github.com/friendica/friendica/pull/7381/commits/23126ee7b41d86b47576f216b4a605ff3a8ad6e3